### PR TITLE
Validation api updates

### PR DESCRIPTION
**Pull request by bedakb**

### CODE REVIEW

File: src/form/form.tsx [hunk 1]
```diff
@@ -400,6 +409,20 @@ class Form<MODEL extends object> extends React.Component<Props<MODEL>, State<MOD
...
+    private getUpdatedFields(prevModel: MODEL, model: MODEL, prevField: string = ''): Array<string> {
+        return flatten(
+            Object.keys(prevModel)
+                .map((field) => {
+                    const newField = prevField ? `${prevField}.${field}` : field;
+                    if (isObject(prevModel[field])) {
+                        return [...this.getUpdatedFields(prevModel[field], model[field], newField)];
+                    }
+                    return prevModel[field] !== model[field] ? newField : [];
+                }, Infinity)
```
Comment (MostafaBalata) [inline src/form/form.tsx]: What is the point of using Infinity here?

Comment (bedakb) [reply src/form/form.tsx]: Ah It should be removed, It's leftover from flatMap fn.

File: src/form/form.tsx [hunk 2]
```diff
@@ -400,6 +409,20 @@ class Form<MODEL extends object> extends React.Component<Props<MODEL>, State<MOD
...
+                .filter(Boolean)
```
[the remark this answers is not in the document]
Comment (bedakb) [reply src/form/form.tsx]: Same, I forgot to remove It.
No need for filtering at all.

File: src/form/form.tsx [hunk 3]
```diff
@@ -203,6 +205,11 @@ class Form<MODEL extends object> extends React.Component<Props<MODEL>, State<MOD
         }
     }
 
+    private validateUpdatedFields() {
+        const fields = this.getUpdatedFields(this.state.initialModel, this.state.model);
+        fields.map(this.validateField);
```
Comment (bedakb) [inline src/form/form.tsx]: Maybe we can use forEach here because we don't care about returned value.

File: src/form/form.tsx [hunk 4]
```diff
@@ -400,6 +409,18 @@ class Form<MODEL extends object> extends React.Component<Props<MODEL>, State<MOD
         }, {});
     }
 
+    private getUpdatedFields(prevModel: MODEL, model: MODEL, prevField: string = ''): Array<string> {
+        return flatten(
```
Comment (BenjaminEckardt) [inline src/form/form.tsx]: Could we test that guy? If necessary just extract this method to a different plan TS file to make it generic and easier to test it.

Comment (bedakb) [reply src/form/form.tsx]: Sure thing, will add tests for this function.

Comment (bedakb) [reply src/form/form.tsx]: @BenjaminEckardt function is extracted and tests are added.
Not sure is It `form` directory good place for this function ?

**Pull request merged**